app/models/pame_evaluation.rb

```diff
@@ -209,64 +191,29 @@ def self.filters_to_json
     ].to_json
   end
 
-  def self.generate_csv(where_statement, restricted_where_statement)
-    where_statement = where_statement.empty? ? '' : "WHERE #{where_statement}"
-    restricted_where_statement = restricted_where_statement.empty? ? '' : "WHERE #{restricted_where_statement}"
-    query = <<-SQL
-      SELECT e.id AS id,
-             e.metadata_id AS metadata_id,
-             e.url AS url,
-             e.year AS evaluation_year,
-             e.methodology AS methodology,
-             protected_areas.wdpa_id AS wdpa_id,
-             ARRAY_TO_STRING(ARRAY_AGG(countries.iso_3),';') AS countries,
-             protected_areas.name AS site_name,
-             designations.name AS designation,
-             pame_sources.data_title AS data_title,
-             pame_sources.resp_party AS resp_party,
-             pame_sources.year AS source_year,
-             pame_sources.language AS language
-             FROM pame_evaluations e
-             INNER JOIN protected_areas ON e.protected_area_id = protected_areas.id
-             LEFT JOIN pame_sources ON e.pame_source_id = pame_sources.id
-             INNER JOIN countries_protected_areas ON protected_areas.id = countries_protected_areas.protected_area_id
-             INNER JOIN countries ON countries_protected_areas.country_id = countries.id
-             INNER JOIN designations ON protected_areas.designation_id = designations.id
-             #{where_statement}
-             GROUP BY e.id, protected_areas.wdpa_id, protected_areas.name, designation, pame_sources.data_title,
-                      pame_sources.resp_party, pame_sources.year, pame_sources.language
-
-             UNION
-
-      SELECT e.id AS id,
-             e.metadata_id AS metadata_id,
-             e.url AS url,
-             e.year AS evaluation_year,
-             e.methodology AS methodology,
-             0 AS wdpa_id,
-             ARRAY_TO_STRING(ARRAY_AGG(countries.iso_3),';') AS countries,
-             e.name AS site_name,
-             NULL AS designation,
-             pame_sources.data_title AS data_title,
-             pame_sources.resp_party AS resp_party,
-             pame_sources.year AS source_year,
-             pame_sources.language AS language
-             FROM pame_evaluations e
-             INNER JOIN pame_sources ON e.pame_source_id = pame_sources.id
-             INNER JOIN countries_pame_evaluations ON e.id = countries_pame_evaluations.pame_evaluation_id
-             INNER JOIN countries ON countries_pame_evaluations.country_id = countries.id
-             #{restricted_where_statement}
-             GROUP BY e.id, wdpa_id, e.name, designation, pame_sources.data_title,
-                      pame_sources.resp_party, pame_sources.year, pame_sources.language;
-    SQL
-    evaluations = ActiveRecord::Base.connection.execute(query)
+  def self.generate_csv(where_params)
```

Comment:
This looks much slower now that is using ActiveRecord. It took at least 30 seconds for me to generate.
You could probably reuse the first bit of the SQL query you deleted and tweak it a little bit to make sure it could also run the `protected_area_id IS NOT NULL` statement.